pkg/internal/controller/controller.go

```diff
@@ -234,7 +234,9 @@ func (c *Controller) reconcileHandler(obj interface{}) bool {
 	// resource to be synced.
 	if result, err := c.Do.Reconcile(req); err != nil {
 		c.Queue.AddRateLimited(req)
-		log.Error(err, "Reconciler error")
+		if log.V(3).Enabled() {
```

Comment:
For successes I understand the change but for errors?

Reply:
There was some discussion on the other PR, to have the error only logged on higher verbosity, chances are that folks using loggers are already logging errors on their own.

Reply:
> hances are that folks using loggers are already logging errors on their own.

Maybe? We don't know that and only the ppl that do use their own logger will speak up about this being too noisy. Also, the general direction we are going is to get ppl to use the integrated loggers or not?

IMHO this is pretty much the definition of an error so increasing the verbosity seems wrong to me.

Reply:
I would be in favor of removing these logs altogether in v0.7.x if we're not ok to use verbosity for an error

@ncdc @detiber thoughts here?

Reply:
Another potential option would be to extend the current use of context w/ logger so that a calling controller can more easily modify the context and have the errors logging here use the logger from the context.

Then we could avoid doing the error logging in the called code (which we currently prefer for the more detailed information it provides), and have the logging here be done with the additional context that is provided by the called reconciler.

That would save having to remember to log errors everywhere on the error path for reconcilers, avoid losing context related to the logging.

Main downside is that it would still potentially log the wrong filename/line number for the error. I'm not sure if there is a better way to handle that generically, though.

Reply:
I guess my point is that we pretty much don't know if and how ppl use these today and I think if we require debug logging to be enabled to see them, then we can also remove those log statements altogether (but probably for 0.7).

> Then we could avoid doing the error logging in the called code (which we currently prefer for the more detailed information it provides), and have the logging here be done with the additional context that is provided by the called reconciler.

So basically tell ppl to put the logger back into the `ctx` if they add additional fields and then re-use that logger here to get these fields?

> Main downside is that it would still potentially log the wrong filename/line number for the error. I'm not sure if there is a better way to handle that generically, though.

Do you have that today? For most (all?) controllers I have worked on there is a function call in the very beginning of the `reconciler` precicely to be able to log all errors. That log line doesn't really have a useful line number associated.

Reply:
My desire/goal/requirement is that when an error is logged, you have the file name and line number **as close to where the error was created as possible**. As things currently stand, this log line emits a file and line number that is about as **far away** from the error's source as possible. I find little to no value in this log message in its current form.

If there is some way for this log invocation to include the error's provenance, then I'm all for keeping it. Otherwise, my vote is to remove it.

Reply:
Removed it, and increased the info log to 5